sdk/include/opentelemetry/sdk/http/http_client.h

```diff
@@ -0,0 +1,153 @@
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace http
+{
+namespace http_api = opentelemetry::http;
+
+static std::function<bool(const std::string &, const std::string &)> CaseInsensitiveComparator =
+    [](const std::string &s1, const std::string &s2) -> bool {
+  std::string str1(s1.length(), ' ');
+  std::string str2(s2.length(), ' ');
+  auto lowerCase = [](char c) -> char { return tolower(c); };
+  std::transform(s1.begin(), s1.end(), str1.begin(), lowerCase);
+  std::transform(s2.begin(), s2.end(), str2.begin(), lowerCase);
+  return str1 < str2;
+};
+
+// HttpHeaders implementation
+class HttpHeaders : http_api::HttpHeaders,
+                    std::multimap<std::string, std::string, decltype(CaseInsensitiveComparator)>
+{
+public:
+  virtual void set(nostd::string_view const &name, nostd::string_view const &value) override
+  {
+    auto range = equal_range(std::string(name.data()));
+    auto hint  = erase(range.first, range.second);
+    insert(hint, std::make_pair(name, value));
+  }
+
+  virtual void add(nostd::string_view const &name, nostd::string_view const &value) override
+  {
+    insert(std::make_pair(name, value));
+  }
+
+  virtual nostd::string_view const &get(nostd::string_view const &name) const override
+  {
+    auto it = find(std::string(name.data()));
+    return (it != end()) ? it->second : m_empty;
+  }
+
+  virtual bool contains(nostd::string_view const &name) const override
+  {
+    auto it = find(std::string(name.data()));
+    return (it != end());
+  }
+
+private:
+  nostd::string_view m_empty{};
+};
+
+class SimpleHttpRequest : public http_api::HttpRequest
+{
+
+public:
+  SimpleHttpRequest(nostd::string_view const &id) : id_(id), method_("GET") {}
+
+  // Gets the HTTP request ID.
+  virtual const nostd::string_view &GetId() const override { return id_; }
+
+  // Sets the request method
+  virtual void SetMethod(nostd::string_view const &method) override { method_ = method; }
+
+  // Sets the HTTP request URI.
+  virtual void SetUrl(nostd::string_view const &url) override { url_ = url; }
+
+  // Gets the HTTP request headers.
+  virtual http_api::HttpHeaders &GetHeaders() const override { return *headers_; }
+
+  // Sets the request body.
+  virtual void SetBody(const uint8_t *const body, const size_t len) override
+  {
+    for (int i = 0; i < len; i++)
+    {
+      body_.push_back(body[i]);
+    }
+  }
+
+  virtual void GetBody(uint8_t *body, size_t &len) override
+  {
+    len = 0;
+    for (auto &e : body_)
+    {
+      body[len++] = e;
+    }
+    len = (len > 0) ? len - 1 : len;
+  }
+
+private:
+  std::unique_ptr<http_api::HttpHeaders> headers_;
+  std::vector<uint8_t> body_;
+  nostd::string_view method_;
+  nostd::string_view id_;
+  nostd::string_view url_;
+};
+
+class SimpleHttpResponse : public http_api::HttpResponse
+{
+
+public:
+  SimpleHttpResponse(nostd::string_view const &id)
+      : id_(id), result_(http_api::HttpResult::HttpResult_LocalFailure), statusCode_(0)
```

Comment:
Wondering whether there is better default status code other than `0`.

Reply:
0 is success so kept it as default. Would use enum type instead of `0` to make it more clear.